### PR TITLE
[fix](move-memtable) make sure index descriptor is set when creating delta writer

### PR DESCRIPTION
**Pull request by kaijchen**
## Proposed changes

Make sure index is found in tablet schema, and make sure index descriptor is set in DeltaWriter request. 

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by kaijchen** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

File: be/src/vec/sink/writer/vtablet_writer_v2.cpp [hunk 1]
```diff
@@ -431,15 +431,27 @@ Status VTabletWriterV2::_write_memtable(std::shared_ptr<vectorized::Block> block
...
                 req.slots = &index->slots;
                 req.schema_hash = index->schema_hash;
+                index_not_found = false;
                 break;
             }
         }
+        if (index_not_found) {
+            LOG(WARNING) << "index " << rows.index_id
+                         << " not found in schema, load_id=" << print_id(_load_id);
+            return static_cast<std::unique_ptr<DeltaWriterV2>>(nullptr);
```
Comment (liaoxin01) [inline be/src/vec/sink/writer/vtablet_writer_v2.cpp:446]:
```suggestion
            return std::unique_ptr<DeltaWriterV2>(nullptr);
```

**Comment by dataroaring** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by liaoxin01 — APPROVED** [review]

**Pull request merged**